src/main/java/rs/teslaris/core/repository/person/OrganisationUnitRepository.java

```diff
@@ -0,0 +1,20 @@
+package rs.teslaris.core.repository.person;
+
+import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import rs.teslaris.core.model.institution.OrganisationUnit;
+
+@Repository
+public interface OrganisationUnitRepository extends JpaRepository<OrganisationUnit, Integer> {
+
+
+    @Query("select ou from  OrganisationUnit ou left join fetch ou.keyword left join fetch ou.name left join fetch ou.researchAreas where ou.id = :id")
+    Optional<OrganisationUnit> findByIdWithLangDataAndResearchArea(Integer id);
```

Comment:
what is the point of this? eager loading instead of lazy loading?

Reply:
It is implemented so we dont have to create a transaction when reading resource, but this makes it problematic because if OU does not have a name assigned to it then it won't return those rows.